File: C/gnomesort/llama-3.1-8b-instant.c
#include <stdio.h>

// Function to implement Gnome Sort
void gnomeSort(int arr[], int n) {
    int i = 1;
    while (i < n) {
        if (arr[i - 1] <= arr[i]) {
            i++;
        } else {
            // Swap elements
            int temp = arr[i - 1];
            arr[i - 1] = arr[i];
            arr[i] = temp;
            if (i > 1) {
                i--;
            }
        }
    }
}

// Function to test the Gnome Sort implementation
void testGnomeSort(int arr[], int n) {
    gnomeSort(arr, n);
    // No need to print the result, just return
}

int main() {
    int arr[1000];
    int size = 0;

    FILE *file = fopen("../../inputs.txt", "r");
    if (!file) {
        return 1;
    }

    while (fscanf(file, "%d", &arr[size]) == 1 && size < 1000) {
        size++;
    }
    fclose(file);

    gnomeSort(arr, size);
    return 0;
}

File: Java/gnomesort/llama_3_1_8b_instant.java

import java.util.ArrayList;
import java.util.List;

public class llama_3_1_8b_instant {

    /**
     * Gnome sort algorithm implementation.
     * 
     * @param arr The array to be sorted.
     */
    public static void gnomeSort(int[] arr) {
        int i = 0;
        while (i < arr.length) {
            if (i == 0 || arr[i] >= arr[i - 1]) {
                i++;
            } else {
                swap(arr, i, i - 1);
                i--;
            }
        }
    }

    /**
     * Swap two elements in the array.
     * 
     * @param arr The array.
     * @param i   The index of the first element.
     * @param j   The index of the second element.
     */
    private static void swap(int[] arr, int i, int j) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }

    public static void main(String[] args) {
        List<Integer> input = new ArrayList<>();

        try (java.io.BufferedReader br = new java.io.BufferedReader(new java.io.FileReader("../../inputs.txt"))) {
            String line;
            while ((line = br.readLine()) != null) {
                input.add(Integer.parseInt(line.trim()));
            }
        } catch (java.io.IOException e) {
            return;
        }

        // Converte List<Integer> diretamente para int[]
        int[] arr = input.stream().mapToInt(Integer::intValue).toArray();

        // Usa o método existente
        gnomeSort(arr);


    }

}
